lib/manager/gradle/build-gradle.ts

```diff
@@ -120,6 +122,22 @@ function moduleStringVariableExpressionVersionFormatMatch(
   );
 }
 
+function groovyPluginVariableVersionFormatMatch(
+  dependency: GradleDependency
+): RegExp {
+  return new RegExp(
```

Comment:
Shouldn't we use `regEx` function from `lib/util/regex` ?

Reply:
Ideally yes. It's essential for user-supplied regex patterns but could be a good idea if we use for all.

Reply:
As the regex contains `dependency.group` i would say it contains user-supplied patterns.

Reply:
Should I also adjust the other uses of RegExp in that file or only change the new ones introduced in this PR?

Reply:
If you could do the others while you're editing this file, that would be great.

Reply:
Unfortunately I seem to have run into a difference in behavior between RegExp and re2.

Assuming a regex that has a single capture group and the following replace call:
```
myString.replace(regex, `$12`)
```
RegExp will replace `$1` by the capture group content since `$12` is not a valid capture group.
re2, however, looks for capture group 12 and failing to find it doesn't replace `$1`, returning the string `$12`.

Do you have any suggestions how to best solve this?
Would using named capture groups be an option since that should be supported by node 10 and re2?

Reply:
re2 does the correct behavior, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace

I think node regex makes something wrong